### PR TITLE
Fix for MAUI RoutedViewHost

### PR DESCRIPTION
**Pull request by ili**
Fix for #3035

**What is the current behavior?**

1. Direct `Router.NavigationStack` manipulations are not affected on MAUI's `NavigationPage.Navigation.NavigationStack`
2. MAUI's `NavigationPage.PoppedToRoot` event is not supported
3. Double navigation to the same `VM` occures on application start

**What is the new behavior?**

Bugs are fixed
Sample app: https://github.com/ili/ReactiveUI-Navigation-Issue/tree/fix-maui-navigation

**Other information**:

Direct `NavigationPage.Navigation` manipulations are used to avoid page activations.

### CODE REVIEW

**Comment by glennawatson** [comment]
Let me test and check out the code for you tomorrow. If it looks good I'll do a release for you then. 

Thanks

File: src/ReactiveUI.Maui/RoutedViewHost.cs
Comment (glennawatson) [inline src/ReactiveUI.Maui/RoutedViewHost.cs:193]: If this might be used by implementors should this be virtual?

Comment (glennawatson) [reply src/ReactiveUI.Maui/RoutedViewHost.cs:193]: Along with the other methods in there

Comment (ili) [reply src/ReactiveUI.Maui/RoutedViewHost.cs]: In general yes, but i'v just copied previous implementation. Mostly I can't see the reason to have method, returning `IObservable<Page>`...

**Comment by ChrisPulman** [comment]
Thanks for this PR. 
I have tested the Android version and this seems to work as expected.

When I tested the same demo on Windows unfortunately the demo displays a blank screen due to a slightly different start-up process. I refactored some of the code to get it working however then I was then faced with a multitude of cross threading issues. 
I will look into this in more detail to see if it's something to do with the manipulation of the core Navigation stack or just a scheduler issue elsewhere.

**Comment by ChrisPulman** [comment]
I can verify that both Windows and Android Targets now operate with the latest commit.
Waiting for @RLittlesII to verify Mac/iOS and to accept the PR

File: src/ReactiveUI.Maui/RoutedViewHost.cs
Comment (ili) [inline src/ReactiveUI.Maui/RoutedViewHost.cs:185]: @ChrisPulman there is the problem with setting title - setting it directly will break later binding.

Also if there are cross-thread issues may be it would be better to do smth like this:


```
			Router?
				.Navigate
				.ObserveOn(RxApp.MainThreadScheduler) // <-- here
				.SelectMany(_ => PagesForViewModel(Router.GetCurrentViewModel()))
				.SelectMany(async page =>

```
?

Comment (ChrisPulman) [reply src/ReactiveUI.Maui/RoutedViewHost.cs:185]: I believe that it will only affect the ability to bind the Title, which with other platforms such as Xamarin we set the Title property too. 
However there could be some argument that we shouldn't be setting this property on any platform to allow the implementor to choose what and how the properties are set or bound. 
There should be commonality across the platforms in my opinion.
I'll ask @glennawatson to add his opinion on this.

Comment (ili) [reply src/ReactiveUI.Maui/RoutedViewHost.cs:185]: @ChrisPulman i can't say about Xamarin, may be there are some difference between Xamarin bindings & MAUI compiled bindings, but the fact is - setting `Title` on navigating breaks bindings. I'v commented this code in my local `RoutedViewHost` implementation, and this saves :)

File: src/ReactiveUI.Maui/RoutedViewHost.cs
Comment (glennawatson) [inline src/ReactiveUI.Maui/RoutedViewHost.cs:214]: This isn't using the inbuilt RxUI way of doing this, so might be better to use RxApp.MainThreadScheduler

**Comment by ili** [comment]
@glennawatson @ChrisPulman really sorry if my comment would be stupid and obvious, but i'v faced with Windows cross thread exceptions until added reference to `ReactiveUI.WinUI` to my project:

```
	<ItemGroup Condition="'$(TargetFramework)' == 'net6.0-windows10.0.19041.0'">
	  <PackageReference Include="ReactiveUI.WinUI">
	    <Version>18.2.5</Version>
	  </PackageReference>
	</ItemGroup>
```

File: src/ReactiveUI.Maui/RoutedViewHost.cs
Comment (ChrisPulman) [inline src/ReactiveUI.Maui/RoutedViewHost.cs:38]: Added Set Title On Navigate Property

File: src/ReactiveUI.Maui/RoutedViewHost.cs
Comment (ChrisPulman) [inline src/ReactiveUI.Maui/RoutedViewHost.cs:75]: Added ObserveOn(RxApp.MainThreadScheduler)

File: src/ReactiveUI.Maui/RoutedViewHost.cs
Comment (ChrisPulman) [inline src/ReactiveUI.Maui/RoutedViewHost.cs:236]: Used RxApp.MainThreadScheduler.Schedule to handle cross threading

**Review by ChrisPulman — COMMENTED** [review]
Waiting for review of @RLittlesII before approval

**Review by glennawatson — APPROVED** [review]

**Pull request merged**
